src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -220,16 +217,12 @@ static SourceGeneratorExecutionVersionMap FilterToProjectCone(SourceGeneratorExe
 
                 // sync over the *info* about all the added/changed projects.  We'll want the info so we can determine
                 // what actually changed.
-                using var _5 = PooledHashSet<Checksum>.GetInstance(out var newChecksumsToSync);
-                newChecksumsToSync.AddRange(newProjectIdToChecksum.Values);
-
-                var newProjectStateChecksums = await _assetProvider.GetAssetsAsync<ProjectStateChecksums>(
-                    assetHint: AssetHint.None, newChecksumsToSync, cancellationToken).ConfigureAwait(false);
-
-                foreach (var (checksum, newProjectStateChecksum) in newProjectStateChecksums)
+                foreach (var (projectId, checksum) in newProjectIdToChecksum)
                 {
-                    Contract.ThrowIfTrue(checksum != newProjectStateChecksum.Checksum);
-                    newProjectIdToStateChecksums.Add(newProjectStateChecksum.ProjectId, newProjectStateChecksum);
+                    var newProjectStateChecksums = await _assetProvider.GetAssetAsync<ProjectStateChecksums>(
+                        assetHint: projectId, checksum, cancellationToken).ConfigureAwait(false);
+                    Contract.ThrowIfTrue(newProjectStateChecksums.Checksum != checksum);
+                    newProjectIdToStateChecksums.Add(projectId, newProjectStateChecksums);
```

Comment:
this also now allows syncing over a project's info without doign a full search for it.